sherpa/astro/tests/test_astro_plot.py

```diff
@@ -25,22 +28,171 @@
 
 from sherpa.astro.data import DataPHA
 from sherpa.astro.plot import SourcePlot, \
-    DataPHAPlot, ModelPHAHistogram
+    DataPHAPlot, ModelPHAHistogram, EnergyFluxHistogram, PhotonFluxHistogram
 from sherpa.astro import plot as aplot
 from sherpa.data import Data1D
 from sherpa.models.basic import Const1D, Gauss1D, Polynom1D
 from sherpa import stats
 from sherpa.utils.err import IOErr
-from sherpa.utils.testing import requires_pylab
 
 
-def test_sourceplot():
+def check_sourceplot_energy(sp, factor=0):
+    """Check for test_sourceplot/test_sourceplot_channel
+
+    Note that the rate setting does not change these tests.
```

Comment:
Is this comment currently applicable?

Reply:
Yes - `test_sourceplot` has `assert data.rate` and `test_sourceplot_counts` has `data.rate = False` and both call `check_sourceplot_energy`. I wanted some note that the rate setting doesn't change these tests since we may want to change this at some point. I orignially added these tests so I could then go and address #1087 but then realized it would take some review and discussion, and it was better to get tests added of the existing functionality.